src/routes/webhooks/customerio.ts

```diff
@@ -193,7 +205,33 @@ export const customerio = async (fastify: FastifyInstance): Promise<void> => {
     },
   });
 
-  fastify.post<ReportingEvent>('/reporting', {
+  fastify.post<WebhookPayload<NotificationPayload>>('/notification', {
+    config: {
+      rawBody: true,
+    },
+    handler: async (req, res) => {
+      const valid = verifyCIOSignature(process.env.CIO_WEBHOOK_SECRET, req);
+      if (!valid) {
+        req.log.warn('cio notifcations webhook invalid signature');
+        return res.status(403).send({ error: 'Invalid signature' });
+      }
+
+      const payload = req.body;
+      if (!validateNotificationPayload(req.body)) {
+        return res.status(400).send({ success: false });
+      }
+
+      try {
+        await sendGenericPush(payload.userIds, payload.notification);
```

Comment:
validate duplicates?

Reply:
Duplicates of what?

Reply:
If userIds, I think OneSignal also handles it on their side

Reply:
yes

Reply:
I've passed it through a set to ensure it's unique.